include/picongpu/fields/incidentField/profiles/BaseParam.hpp

```diff
@@ -96,7 +96,35 @@ namespace picongpu
                             return 0.0;
                         }
 
+                        /** SFINAE detection if the user parameter define the variable TIME_DELAY_SI

```

Comment:
What is: `SFINAE` ?

Reply:
SFINAE stands for [Substitution failure is not an error](https://en.wikipedia.org/wiki/Substitution_failure_is_not_an_error) which I use here to overload the template parameters of this function, which then makes the time delay optional by choosing the best fitting function to call. However, I copied this comment from `Functors.hpp` where this approach is also used to make the `Origin` optional. When copying, I tried to fix [the typo in "dedection"](https://github.com/ComputationalRadiationPhysics/picongpu/blob/7337a134fe7c2176f1e804a5385a2d3c467b8092/include/picongpu/fields/incidentField/Functors.hpp#L116) in the original code to "detection"... but looking at it now: Should it be "deduction"? @psychocoderHPC

Reply:
Yes should be deduction!